addons/spreadsheet/static/src/global_filters/helpers.js

```diff
@@ -129,7 +129,7 @@ export function checkFilterValueIsValid(filter, value) {
  * @returns {boolean}
  */
 function isTextFilterValueValid(value) {
-    return Array.isArray(value) && value.every((text) => typeof text === "string");
+    return Array.isArray(value) && value.length && value.every((text) => typeof text === "string");
```

Comment:
shouldn't `isTextFilterValueValid(undefined)` return true ? It's a valid value no ?